consensus/pacemaker/module.go

```diff
@@ -243,6 +265,87 @@ func (m *pacemaker) NewHeight() {
 		)
 }
 
+// This is called each time there is a NewRound message received by the leader from replicas
+// With the introduction of MinBlockTimeMsec delay, multiple concurrent calls may happen
+// It makes sure that:
+//   - Block proposal is made by only one of the possible `HotstuffLeaderMessageHandler.HandleNewRoundMessage()` concurrent (because delayed) calls
+//   - If the timer expires, the first call to this method will trigger the block proposal
+//   - If a late message is received after a block is proposed by another call, the late message is discarded
+//   - Reads and affectations to pacemaker.latestPrepareRequest state are protected by a mutex
+func (m *pacemaker) ProcessDelayedBlockPrepare(currentHeight uint64) chan bool {
```

Comment:
Does this need to be publically exposed?


If so, make sure it's part of the pacemaker interface under the `shared/` module?

Reply:
Yes, it's basically called from the consensus module. Will make sure `shared/` is aligned

Reply:
Updating the interface made some errors to raise at `consensus/module_consensus_pacemaker.go#L12`
```go
var _ modules.ConsensusPacemaker = &consensusModule{}
```

Which makes me question the assignment :thinking:

Reply:
`ProcessDelayedBlockPrepare` should only be used by `Consensus` module (specifically `hotstuff_leader`, and maybe `hotstuff_replica`). Since the PaceMaker is acting like a submodule to `Consensus`, there is nothing to expose through the `Bus`.